src/components/DatePicker/__tests__/DatePicker.spec.tsx

```diff
@@ -0,0 +1,104 @@
+import React from 'react';
+import { render, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import '@testing-library/jest-dom';
+import MockDate from 'mockdate';
+import { DatePicker } from '../DatePicker';
+
+describe('DatePicker', () => {
+    let view;
+    const handleChange = jest.fn();
+
+    const minDate = new Date('2021-11-23');
+    const maxDate = new Date('2023-03-04');
+
+    const openCalendar = async (user) => {
+        await user.click(screen.getByRole('textbox'));
+    };
+
+    beforeAll(() => {
+        MockDate.set('2021-12-24T13:00:00.000Z');
+    });
+
+    afterAll(() => {
+        MockDate.reset();
+    });
+
+    beforeEach(() => {
+        view = render(<DatePicker onChange={handleChange} minDate={minDate} maxDate={maxDate} />);
+    });
+
+    it('should render correctly', () => {
+        expect(view.container).toMatchSnapshot();
+        expect(screen.getByRole('textbox')).toBeInTheDocument();
+    });
+
+    it('should render calendar header correctly when calendar is open', async () => {
+        const user = userEvent.setup();
+        await openCalendar(user);
+        // taking full snapshot of the calendar fails in CD/CI.
```

Comment:
I don't get it. I've checked pipeline and it's passed

Reply:
Looks like the CD/CI failed when we added a snapshot of the calendar with enzyme. My question was if the same still happens with RTL. If not we can include the snapshot in the test, otherwise just leave the comment here as it is.

Reply:
Added toMatchSnapshot. But what about these:
    // Some small time difference in dates - needs further investigation.
        // Since testing 3rd party library is not a good practice anyway,
        // we will just check that our custom header is correct
        // screen.getByRole('option');
I've removed them and added some cases. If i understand them correct.

Reply:
About those I don't know.